core/src/main/scala/kafka/log/Log.scala

```diff
@@ -139,7 +140,13 @@ class Log(val dir: File,
   /* Load the log segments from the log files on disk */
   private def loadSegments() {
     // create the log directory if it doesn't exist
-    dir.mkdirs()
+    if (!Files.exists(dir.toPath())) {
+      try {
+          Files.createDirectory(dir.toPath())    
+      } catch {
+        case e: IOException => throw new KafkaException("Error in creating new directory.", e)
```

Comment:
Since we are wrapping the IOException in a KafkaException I think it may make sense to add the Kafka context to the error message. Something like:

```scala
throw new KafkaException("Error creating log directory ${dir.toPath}", e)
```